src/utility_components/NumericField.jsx

```diff
@@ -0,0 +1,96 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { forceNumber } from '../utils';
+
+export class NumericField extends Component {
+    constructor(props) {
+        super(props);
+
+        this.onFocus = this.onFocus.bind(this);
+        this.onBlur = this.onBlur.bind(this);
+        this.onChange = this.onChange.bind(this);
+
+        this.state = {
+            value: this.props.value,
+            fieldValue: this.props.value,
+        };
+    }
+    static getDerivedStateFromProps(props, state) {
+        if (props.value !== state.value) {
+            return {
+                value: props.value,
+                fieldValue: props.value,
+            };
+        } else {
+            return null;
+        }
+    }
+    onFocus(e) {
+        e.target.select();
+    }
+    onBlur(e) {
+        // Validate and push state up here
+        let val = forceNumber(e.target.value);
+        if (val >= this.props.min && val <= this.props.max) {
+            if (this.props.onChange) {
+                this.props.onChange(val);
+            }
+            /* eslint-disable */
+            console.log('setting the state');
+            this.setState({
+                value: val,
+                fieldValue: val,
+            });
+        }
+    }
+    onChange(e) {
+        // track local changes here
+        e.preventDefault();
+        this.setState({
+            fieldValue: e.target.value,
```

Comment:
Convert `e.target.value` with `forceNumber()` here, because it might be a string